core/templates/pages/contributor-dashboard-page/modal-templates/translation-modal.component.ts

```diff
@@ -0,0 +1,191 @@
+  ) {}
+
+  ngOnInit(): void {
+    this.activeLanguageCode =
+    this.translationLanguageService.getActiveLanguageCode();
+    // We need to set the context here so that the rte fetches
+    // images for the given ENTITY_TYPE and targetId.
+    this.contextService.setCustomEntityContext(
+      AppConstants.ENTITY_TYPE.EXPLORATION, this.opportunity.id);
+    this.subheading = this.opportunity.subheading;
+    this.heading = this.opportunity.heading;
+    this.contextService.setImageSaveDestinationToLocalStorage();
+    this.languageDescription = (
+      this.translationLanguageService.getActiveLanguageDescription());
+    this.translateTextService.init(
+      this.opportunity.id,
+      this.translationLanguageService.getActiveLanguageCode(),
+      () => {
+        const textAndAvailability = (
+          this.translateTextService.getTextToTranslate());
+        this.textToTranslate = textAndAvailability.text;
+        this.moreAvailable = textAndAvailability.more;
+        this.loadingData = false;
+      });
+    this.HTML_SCHEMA = {
+      type: 'html',
+      ui_config: {
+        hide_complex_extensions: true,
+        language: this.translationLanguageService.getActiveLanguageCode(),
+        languageDirection: (
+          this.translationLanguageService.getActiveLanguageDirection())
+      }
+    };
+  }
+
+  close(): void {
+    this.activeModal.close();
+  }
+
+  getHtmlSchema(): HTMLSchema {
+    return this.HTML_SCHEMA;
+  }
+
+  onContentClick(event: MouseEvent): void {
+    if (this.isCopyModeActive()) {
+      event.stopPropagation();
+    }
+    this.ckEditorCopyContentService.broadcastCopy(event.target as HTMLElement);
+  }
+
+  isCopyModeActive(): boolean {
+    return this.ckEditorCopyContentService.copyModeActive;
+  }
+
+  updateHtml($event: string): void {
+    if ($event !== this.activeWrittenTranslation.html) {
+      this.activeWrittenTranslation.html = $event;
+      this.changeDetectorRef.detectChanges();
+    }
+  }
+
+  skipActiveTranslation(): void {
+    const textAndAvailability = (
+      this.translateTextService.getTextToTranslate());
+    this.textToTranslate = textAndAvailability.text;
+    this.moreAvailable = textAndAvailability.more;
+    this.activeWrittenTranslation.html = '';
+  }
+
+  returnToPreviousTranslation(): void {
+    const textAndAvailability = (
+      this.translateTextService.getPreviousTextToTranslate());
+    this.textToTranslate = textAndAvailability.text;
+    this.previousTranslationAvailable = textAndAvailability.more;
+  }
+
+  suggestTranslatedText(): void {
+    if (!this.uploadingTranslation && !this.loadingData) {
+      this.siteAnalyticsService
+        .registerContributorDashboardSubmitSuggestionEvent('Translation');
+      this.uploadingTranslation = true;
+      const imagesData = this.imageLocalStorageService.getStoredImagesData();
+      this.imageLocalStorageService.flushStoredImagesData();
+      this.contextService.resetImageSaveDestination();
```

Comment:
Hi @ashutoshc8101 can you please explain why this placed here? It seems incorrect compared to the directive file. This has resulted in a regression.

Reply:
Fixed it in https://github.com/oppia/oppia/pull/12573.

Reply:
@kevintab95 When resolving merge conflicts, I was not able understand your changes (as this PR was originally from another author) and was not sure of the side effects it would have on my PR. At that point, this PR was approved and ready to be merged. I didn't want to cause any regressions by wrongly resolving merge conflicts. So, I tried to play safe, by leaving the code as it was (undoing your changes). I apologize for the inconvenience casused.

Reply:
I should have contacted you. I apologize for the inconvenience caused.

Reply:
Thanks @ashutoshc8101 for explaining. I think there are a few problematic patterns though. So I'd like to address them individually. RIL

> 1. This PR was originally from another author, so, I had issues understanding some sections of this PR.

If you are taking over a PR, make sure you completely understand the changes in the PR and what needs to be tested. Always get clarity from the original author first but also re-do the testing side of things by yourself (there may have been gaps in what the original author tested/wrote). If you have trouble understanding sections of a PR, talk to the original author or reach out to the relevant team leads, they can help. Ignoring / not understanding sections of code in your own PR is not great and must be avoided -- you are expected to be responsible for your changes.

> 2. When resolving merge conflicts, I was not sure what and how your changes were fixing certain issues and how they would have side effects on this PR.

In general, if you have doubts when resolving merge conflicts, you can identify the PR conflicting PR using [blame](https://docs.github.com/en/github/managing-files-in-a-repository/tracking-changes-in-a-file) on Github and read through the description or reach out to the author to understand the changes better. This is why we require that PR descriptions give a summary of what the change is and why it was necessary.

In this particular case, I have explained what was the issue in the description [here](https://github.com/oppia/oppia/pull/12439#issue-609378869) and mentions affected area (i.e. reviewing several translations with images in the translation suggestion modal). Let me know if you have questions about this. I'm happy to edit the description to be more clear if you think it would help future contributors referencing the PR.

> 3. I was having issues migrating your changes to angular.

The only issue I faced was related to a duplicate context service being present which had a simple fix that @srijanreddy98 recommended. If you run into this sort of a situation, please escalate with a team lead. Please avoid merging changes you are unsure about or know can lead to issues later on.

> 4. Sean had informed me that this PR is important and should be merged fast.

This is not a valid reason to ignore issues in the code that can result in regressions. At the very least, an immediate follow-up is required. You should have left a TODO in the code and followed up with the conflicting PR author to get the issue fixed after this PR was merged.

> 5. Your changes were very small compared to this PR.

This is not a valid reason to undo changes (big or small) on develop. The bug that the PR fixed was something reported by real users on the server. If this code would have been pushed to the server, it would have seriously impeded translation of our lessons and resulted in wasted hours trying to identify/fix/test/deploy -- we definitely want to avoid this.

> 6. I was sure that you will identify this undo in the future and will fix it.

No. Don't do this. I found this only by chance as I was fixing an unrelated issue in the same feature and found myself blocked. Do not introduce regressions thinking someone would catch and fix it. As mentioned above, if there is a trade-off and you need to undo something (this is very rare), you will need to at least inform the relevant author so an immediate follow-up fix can be made.

> 7. This PR was at the post-approval state at that time and any changes I would make will go directly to develop.
> So, I thought of undoing your changes, this way I could merge this PR fast without being un sure of the side effects of post approval changes.

I don't see why this would be an issue if you were only fixing the merge conflict. Merge commits are not tedious or problematic to review -- Github handles the diff view correctly for these. Also, "merge this PR fast" is a good attitude to have but it is not great to ignore issues in the code because it affects our end users.

Reply:
@kevintab95 Thanks for your guidance. The PR description you have provided is sufficient, my ignorance is at fault. The main reason behind this was only that I was afraid of pushing incorrect code post-approval. The original author was not online, that's why I took this up.

Reply:
I think **undo** is not the right word to use here, you fixed the angularjs directive and I didn't add the corresponding fix to the angular component.

Reply:
Well, yeah in a way. When migrating you copy the old logic and your migration did not include the current logic. But not including is basically undo-ing a fix (functionality-wise).

Reply:
On a lighter note, The way I see the migration is, **understanding the goals achieved by the old logic and then writing code in angular to achieve the same goal**. Copying does work for simple directives (90% cases) but for complex directives like base-content, schema-based-editor, understanding is a necessity. Sometimes the means which were available in `anguarjs` like $compile and `uib-tabs`, are not available in angular, so, we have to find alternatives. The migration process is a mix of both copying and understanding, I copy the old logic (to speed up the process) until it does not work and a new way has to be found. Yeah, when migrating we copy the old logic but it is not the only thing that we do. 

Please take this on a lighter note. I just want to put my point here. Please ignore if this seems irrelevant.

Reply:
I'm in agreement that it is not merely a copy-paste, there may be pre-work required and it can be tedious (so I think this is great work and thanks a lot for doing it!). But in the end, the functionality must remain the same to a user -- this is very important.

Also, as you have already pointed out, this PR seems to be a case where there was no drive to understand or address the underlying logic correctly. This was my only concern - you can address this by following the tips I gave above. Again, I want to stress that this is in no way undermining your migration efforts (in fact I really admire your contributions and I'm sure the rest of the community does too!) - many contributors face these issues when starting off and it is great that you have identified the issue early on and can fix it in future PRs!

Reply:
I agree that my approach with this PR was not right.

Reply:
The reasons I have mentioned above are not justifications. The decision to undo the changes was wrong. I just wanted the highlight the factors which have contributed to this human error from my side.